src/providers/zerion.rs

```diff
@@ -67,11 +72,22 @@ impl HistoryProvider for ZerionProvider {
         &self,
         address: String,
         body: Bytes,
+        params: HistoryQueryParams,
     ) -> RpcResult<HistoryResponseBody> {
-        let uri = format!(
-            "https://api.zerion.io/v1/wallets/{}/transactions/?currency=usd",
-            address
-        );
+        let currency = params.currency.unwrap_or("usd".to_string());
```

Comment:
Avoid unnecessary allocation

```suggestion
        let currency = params.currency.as_ref().unwrap_or("usd");
```